### PR TITLE
Switch: Update for libnx 4.0 (hid refactor)

### PR DESCRIPTION
**Pull request by carstene1ns**
- Only tested in emulator (yet).
- needs toolchain update

### CODE REVIEW

File: src/platform/switch/switch_ui.cpp [hunk 1]
```diff
@@ -384,7 +390,7 @@ void NxUi::UpdateDisplay() {
 
 	// full res when docked
 	int width = 1280, height = 720;
-	if (appletGetOperationMode() == AppletOperationMode_Docked) {
+	if (appletGetOperationMode() == AppletOperationMode_Console) {
```
Comment (Ghabry) [inline src/platform/switch/switch_ui.cpp:393]: "Lets rename some random enums" 🙄

**Comment by Ghabry** [comment]
Thanks for the patch! Now we only must find testers :/

**Comment by Ghabry** [comment]
@carstene1ns 
Can you provide a ready to install file for Switch? @elsemieni can test it :+1:

**Review by Ghabry — COMMENTED** [review]
Code wise this looks fine. Just waiting for the hardware test :+1:

**Comment by carstene1ns** [comment]
Here is the hardware test: https://youtu.be/TQ59T_cmJOY
As expected, it works fine (:grinning:)
Will push a small `Makefile` change yet, then can be merged.

**Review by fdelapena — APPROVED** [review]

**Comment by carstene1ns** [comment]
Will push a small buildscripts update and then only needs a toolchain rebuild (again).

**Pull request merged**